### PR TITLE
Tool for migration assets to screenly-pro version

### PR DESCRIPTION
**Pull request by rusko124**
Discussion about this tool #930 

This tool-script is written in python and and helps to transfer assets from Screenly-OSE to Screenly-PRO.
[Click](https://click.palletsprojects.com/en/5.x/) library used for CLI interface.
[Ngrok](https://ngrok.com/) used for serving the assets files locally.

### CODE REVIEW

File: ansible/roles/tools/files/nginx.conf [hunk 1]
```diff
@@ -0,0 +1,6 @@
+server {
+    listen         8000;
```
Comment (vpetersson) [inline ansible/roles/tools/files/nginx.conf]: should be `127.0.0.1:8000;`

File: ansible/roles/tools/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+- name: Installs nginx config
+  copy:
+    src: nginx.conf
+    dest:  /etc/nginx/sites-enabled/screenly_assets.conf
+    mode: 644
+    owner: root
+    group: root
+
+- name: Add custom-path to /etc/profile.d.
```
Comment (vpetersson) [inline ansible/roles/tools/tasks/main.yml]: Why is this required?

File: ansible/roles/tools/files/nginx.conf (hunk 1 above)
Comment (rusko124) [reply ansible/roles/tools/files/nginx.conf]: Got it!

File: ansible/roles/tools/tasks/main.yml (hunk 2 above)
Comment (rusko124) [reply ansible/roles/tools/tasks/main.yml]: In order to use `sh.ngrok`
What do you think about this?

Comment (vpetersson) [reply ansible/roles/tools/tasks/main.yml]: I'd recommend that you refactor this to either:
 * Use the `ngrok-client` apt package (if available)
 * Copy it to `/usr/local/bin/`

File: tools/assets-migration-to-screenly-pro.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+    while True:
+        if try_count >= try_connection:
+            raise Exception('Failed start ngrok')
+        try:
+            requests.get('http://127.0.0.1:%i' % PORT_NGROK, timeout=10)
+            break
+        except requests.exceptions.ConnectionError:
+            try_count += 1
+            time.sleep(0.1)
+            continue
```
Comment (korvyashka) [inline tools/assets-migration-to-screenly-pro.py]: No reason for continue here.

File: tools/assets-migration-to-screenly-pro.py [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+token = None
+ngrok_public_url = None
+
+
+################################
+# Suprocesses
+################################
+
+def start_http_ngrok_process(try_connection=100):
+    global ngrok_process
```
Comment (korvyashka) [inline tools/assets-migration-to-screenly-pro.py]: It seems ngrok_process is not used outside of this func - is it that needed for it to be global?

File: tools/assets-migration-to-screenly-pro.py [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+    click.echo(click.style("Ngrok starting ...", fg='yellow'))
+    ngrok_process = sh.ngrok('http', str(PORT), _bg=True, _in=os.devnull, _out=os.devnull, _err=sys.stderr)
+    try_count = 0
+    while True:
```
Comment (korvyashka) [inline tools/assets-migration-to-screenly-pro.py]: Just a note - could you add spacing between blocks of code to increase readability?

File: tools/assets-migration-to-screenly-pro.py [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+        if try_count >= try_connection:
+            raise Exception('Could not take a public url ngrok')
+        response = requests.get('http://127.0.0.1:%i/api/tunnels' % PORT_NGROK, timeout=10).json()
+        if response['tunnels']:
+            break
+        else:
+            try_count += 1
+            time.sleep(0.1)
```
Comment (korvyashka) [inline tools/assets-migration-to-screenly-pro.py]: Probably instead of doing lots of requests linearly - may be it can be less but non-linearly?

File: tools/assets-migration-to-screenly-pro.py [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+        password = click.prompt('Password', hide_input=True)
+        auth = HTTPBasicAuth(login, password)
+    else:
+        auth = None
+    response = requests.get(ASSETS_SCREENLY_OSE_API, timeout=10, auth=auth)
+    if response.status_code == 200:
+        return response.json()
+    elif response.status_code == 401:
+        raise Exception('Access denied')
+    else:
```
Comment (korvyashka) [inline tools/assets-migration-to-screenly-pro.py]: No need of else here.

File: tools/assets-migration-to-screenly-pro.py [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+    asset_uri = asset['uri']
+    if asset_uri.startswith(HOME):
+        asset_uri = os.path.join(ngrok_public_url, asset['asset_id'])
+    data = {
+        'title': asset['name'],
+        'source_url': asset_uri,
+        # 'duration': asset['duration'] #CURRENTLY DOESN'T WORK
+    }
+    response = requests.post(endpoind_url, data=data, headers=headers)
+    if response.status_code == 200:
```
Comment (korvyashka) [inline tools/assets-migration-to-screenly-pro.py]: it can be just "return response.status_code == 200"

**Comment by vpetersson** [comment]
Thanks!
